Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/BaseDAO.java

```diff
@@ -123,6 +123,24 @@ public void remove(T baseBean) throws DAOException {
         return (List<T>) session.createQuery(query).list();
     }
 
+    @SuppressWarnings("unchecked")
+    List<Double> getAverage(String field, String query) {
+        Session session = HibernateHelper.getHibernateSession();
+        return (List<Double>) session.createQuery("SELECT AVG(" + field + ") " + query).list();
+    }
+
+    @SuppressWarnings("unchecked")
+    List<Long> getCount(String field, String query) {
```

Comment:
How can a SQL count() method return more then one value?

Reply:
There are possibly many values with usage of Group By clausule.